### ISSUE TITLE
Glimpse should identify if the system is using a dark or light theme and switch automatically

### ISSUE THREAD

**Issue by TrechNex** [glimpse-editor/Glimpse#177]
Hopefully this turns out to be more practical than #91 was!

Currently we default to the dark theme because that is what upstream does, but it would be cool if Windows users on the Light theme had their UI automatically use the right one without having to switch.

I do not know how practical this is for Linux-based systems. Glimpse uses its own theme engine because we are still using a mix of GTK2 and GTK3.

**Comment by TrechNex** [comment]
The other thing to think about is whether this should override the user-set choice, or if there should be an "off" switch for this.

**Comment by Member1221** [comment]
On that front it'd require us to hook in to win32 apis, adding that to the automake dependency would be p tedious.

**Comment by bochecha** [comment]
I don't know about Windows, but on Linux there is work going on between GNOME, elementary and a few others, to try and have such a "system dark mode" so apps could automatically switch.

I strongly suggest waiting for that to happen, to avoid doing the work twice.

https://discourse.gnome.org/t/gtk-adwaita-and-vendor-styles/1641
https://blog.elementary.io/the-need-for-a-freedesktop-dark-style-preference/

**Comment by bochecha** [comment]
> The other thing to think about is whether this should override the user-set choice, or if there should be an "off" switch for this.

The way it's likely going to work on GNOME/Elementary is there would be a system setting to "prefer a dark mode". Then apps would opt-in to respect that setting, if app devs decided to make it possible and, did the work and tested it.

That eventual system setting is already "user-set choice".

**Comment by Member1221** [comment]
windows and macOS already have options for such, so yeah we should implement this at some point after GNOME implements their system.

**Comment by TrechNex** [comment]
This looks like something that we will inherit when we move to upstream v3.0. Leaving as a "fork candidate".

**Comment by TrechNex** [comment]
Closing as this is never likely to be implemented.
